Documentation/Evolution/DelimiterSyntax.md

```diff
@@ -0,0 +1,165 @@
+
+### Single letter `r'...'`
+
+We could choose to shorten the literal prefix to just `r`. However this could potentially be confused to mean "raw", especially as Python uses this syntax for raw strings.
+
+### Forward slashes `/.../`
+
+Forward slashes are a regex term of art, and are used as the delimiters for regex literals in Perl, JavaScript and Ruby. However, they would be an awkward fit in Swift's language grammar, and would not provide a path for extensibility.
+
+#### Parsing ambiguities
+
+The primary parsing ambiguity with `/.../` delimiters is with comment syntax.
+
+An empty regex literal would conflict with line comment syntax `//`. While this isn't a particularly useful thing to express, it may lead to an awkward user typing experience. In particular, as you begin to type a regex literal, as it could form a comment before you start typing the contents.
+
+Line comment syntax additionally means that a potential multi-line version of a regular expression literal would not be able to use `///` delimiters, in accordance with the precedent set by multi-line string literals `"""`. However it may potentially be possible to use a raw delimiter syntax e.g `#/` instead for both raw and multiline. This would come with a slightly suboptimal user experience, as a user beginning to type a raw regex literal would have everything else in the file treated as regex syntax until they insert the closing delimiter.
+
+There is also a conflict with block comment syntax, when surrounding a regex literal ending with `*`, for example:
+
+```swift
+/*
+let regex = /x*/
+*/
+```
+
+In this case, the block comment would prematurely end on the second line, rather than extending all the way to the third line as the user would expect. This is already an issue today with `*/` in a string literal, however it is much more likely to occur in a regular expression given the prevalence of the `*` quantifier.
+
+Block comment syntax also means that a regex literal would not be able to start with the `*` character, however this is less of a concern as it would not be valid regex syntax.
+
+Finally, there would be a minor ambiguity with infix operators used with regex literals. When used without whitespace, e.g `x+/y/`, the expression will be treated as using an infix operator `+/`. Whitespace is therefore required `x + /y/` for regex literal interpretation.
+
+#### Language changes required
+
+In addition to ambiguities listed above, there are also some parsing ambiguities that would require the following language changes:
+
+- Deprecation of prefix operators containing the `/` character.
+- Changing parsing rules such that `/` starting a new line forms a regex literal, which may break source for operator chains over multiple lines.
+
```

Comment:
A huge point is that in a result builder, such as the regex DSL for whom intermixing regex literals is a high-impact feature, slash would continue an expression instead of define a new one. Would we treat `re'...'` differently, since it's clearly not a binary operator?